packages/cozy-clisk/src/launcher/saveFiles.js

```diff
@@ -157,6 +161,44 @@ const saveFiles = async (client, entries, folderPath, options) => {
   return savedEntries
 }
 
+/**
+ * Ensure the existence of all destination folders : folderPath, options.subPath and all subPaths
+ * which may be present in each entries.
+ * If the user changes some folders during the execution of saveFile, we will catch and fix the
+ * errors.
+ */
+async function ensureAllDestinationFolders({
+  client,
+  entries,
+  folderPath,
+  options
+}) {
+  const folderMap = new Map()
+  const fileCollection = client.collection('io.cozy.files')
+
+  const pathsList = [folderPath]
+
+  // construct an Array with all the paths to ensure the existence of
+  if (options.subPath) {
+    pathsList.push(folderPath + '/' + options.subPath)
+  }
+  const notFilteredEntriesPathList = entries
+    .map(entry => (entry.subPath ? folderPath + '/' + entry.subPath : false))
+    .filter(Boolean)
+  const entriesPathList = Array.from(
+    new Set(notFilteredEntriesPathList.filter(Boolean))
+  )
+  pathsList.push(...entriesPathList)
+
+  // now ensure paths and put their ids in folderMap
+  for (const path of pathsList) {
+    const createPathResult = await fileCollection.createDirectoryByPath(path)
+    folderMap.set(path, createPathResult.data._id)
+  }
+
+  return folderMap
+}
```

Comment:
Can't you use https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/FileCollection.js#L865-L871 for instance?